static/docs/commands-reference/pull.md

```diff
@@ -1,53 +1,116 @@
 # pull
 
-Pulls data files to the working space.
+Pulls data files to the local workspace from a remote DVC cache based on relative changes between the two.
 
-The set of data files to pull (usually it means downloading from the remote
-storage if file is not in the local cache yet) is determined by analyzing all
-`.dvc` files in the current branch, unless `--all-branches` is specified.
+## Usage
 
-The command pulls only output of a specific stage if dvc file is specified `dvc pull data.zip.dvc`.
+```usage
+    usage: dvc pull [-h] [-q | -v] [-j JOBS]
+                    [--show-checksums] [-r REMOTE]
+                    [-a] [-T] [-d] [-f] [-R]
+                    [targets [targets ...]]
+
+    Pull data files from the cloud.
+
+    positional arguments:
+      targets               DVC files.
+```
+
+## Description
+
+DVC supports sharing a data cache between workspaces.  Members of a team may
+be working with the same data, but using different algorithms, and will find
+it useful to share data.  The `dvc pull` command is analogous to `git pull`,
+and allows one to retrieve data from a remote cache.
+
+For an overview of the process of sharing data between DVC workspaces and remote
+caches, see:
+[Share Data And Model Files](/doc/use-cases/share-data-and-model-files)
+
+Using the command `dvc status --remote REMOTE` determines whether the local
+cache has differences from the named remote cache.
+
+The value for `REMOTE` is a cache name defined using the `dvc remote` command.
+If no REMOTE is given, an error message is printed.  If the `--remote REMOTE`
+option is not specified, then the default remote, configured with the
+`core.config` config option, is used.  See `dvc remote`, `dvc config` and
+[remote storages](/doc/get-started/configure) for more information on how to
+configure remote storage.
+
+With no arguments, just `dvc pull` or `dvc pull --remote REMOTE`, it
+downloads only the changed files corresponding to the currently checked-out DVC
```

Comment:
I'm not sure that `changed files` is the correct term. It just pulls missing references, missing files, etc. The same for -R option and its description.

Reply:
This is the same issue as in other comments - that I did not understand enough that it is simply looking for missing files.  I've reworded this section appropriately.